code/raw-trie.lisp

```diff
@@ -0,0 +1,148 @@
+(cl:in-package #:spell)
+
+;;; `raw-node' class
+
+(defclass raw-node (node) ())
+
+(defmethod %insert ((object t) (string string) (suffix (eql 0)) (node raw-node))
+  (change-class node 'raw-leaf-node)
+  (%insert object string 0 node))
+
+(defmethod %insert ((object t) (string string) (suffix integer) (node raw-node))
+  (change-class node 'raw-interior-node)
+  (%insert object string suffix node))
+
+;;; `raw-leaf-mixin' class and leaf node protocol methods
+
+(defclass raw-leaf-mixin (leaf-mixin)
+  ((%entries :type   simple-vector
+             :writer (setf %entries))))
+
+(defmethod utilities.print-items:print-items append ((object raw-leaf-mixin))
+  (let ((entries '()))
+    (map-entries (lambda (entry)
+                   (push (class-name (class-of entry)) entries))
+                 object  (%entries object))
+    `(((:entries (:after :children)) " entries: ~{~A~^ ~}" ,entries))))
+
+(defmethod map-entries
+    ((function function) (node raw-leaf-mixin) (entries vector))
+  (declare (type simple-vector entries))
+  (map nil function entries))
+
+(defmethod add-entry ((entry t) (node raw-leaf-mixin) (entries vector))
+  (declare (type simple-vector entries))
+  (let ((new-entries (make-array (+ 1 (length entries)))))
+    (setf (aref new-entries 0)   entry
+          (subseq new-entries 1) entries)
+    new-entries))
+
+#-minimal-raw-trie
+(defmethod %lookup
+    ((function function) (string string) (suffix (eql 0)) (node raw-leaf-mixin))
+  (map-entries
+   (lambda (word)
+     ;; This is temporary: make an instance of the explicit base
+     ;; "sibling" class.
+     (let ((word (if (typep word 'implicit-base-mixin)
+                     (let* ((class-name  (class-name (class-of word)))
+                            (class-info  (find class-name *word-classes*
+                                               :test #'eq :key #'third))
+                            (new-class   (find-class (fourth class-info)))
+                            (base-suffix (base-suffix word))
+                            (base        (if (plusp base-suffix)
+                                             (subseq string 0 (- (length string)
+                                                                 base-suffix))
+                                             string)))
+                       (if (slot-exists-p (c2mop:class-prototype new-class) '%info)
+                           (let ((new-info (ldb (byte 29 +base-suffix-bits+)
```

Comment:
`29` is a magic number here, can it get a better description?

Reply:
Nice, I thought, I caught all of those. I will probably have to shuffle some stuff around to make the information available at that point.

Reply:
Done.